mautic/mautic
# POC for the composerisation of Mautic TPROD-147

#### Description:
This PR moves the plugins out of the codebase and allows Mautic to be installed with different ways. This PR ideally doesn't change the way people can install mautic, but it offers possibilities such as 

`composer create-project mautic/recommended-project:4.x-dev mauticcomposerproject --no-interaction`

What is still open before this can get green light

- the .htaccess files need to be added to vendor if it is installed from mautic/mautic instead of mautic/recommended-project. See also https://github.com/drupal/core-file-security/tree/9.2.x how it is done there.
- The themes are still in the main repository, this is fine for now, but it is blocking for the users of the recommeded-project part. Ideally the themes live in app/themes and themes are also allowed in themes/ or wherever custom themes are placed. This is different than plugins as plugins are always optional, themes are not.
- Many image files are not referenced correctly yet. The references still go to media/images, they should go to app/assets/images 
-  https://github.com/composer/installers/pull/484 should be committed, then we remove the complex override of this repository and the file gets a lot simpler.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. See if Mautic still works as expected
3. Run composer update --with-dependencies, see if mautic still works as expecred
